src/controllers/IngredientController.py

```diff
@@ -0,0 +1,62 @@
+from flask import (Blueprint, jsonify, request)
+from src.models.Ingredient import Ingredient
+
+from src.services.IngredientService import IngredientService
+
+
+bp = Blueprint("ingredients", __name__, url_prefix="/ingredients")
+service = IngredientService()
+pass
+
+
+@bp.route("/", methods=["GET"])
+def get_all_ingredients():
+    data = service.getAllIngredients()
+    return jsonify({
+        "status": "Success",
+        "data": {
+            "ingredients": data
+        }
+    }), 200
+
+
+@bp.route("/ingredient", methods=["POST"])
+def add_ingredient():
+    json_data = request.json
+    ingredient: Ingredient = Ingredient(
+        0, json_data["name"], json_data["unit"], json_data["available"])
+    id = service.addIngredient(ingredient)
+    ingredient.id = id
+    return jsonify({
+        "status": "Success",
+        "data": {
+            "ingredient": ingredient.serialize()
+        }
+    }), 200
+
+
+@bp.route("/ingredient", methods=["PUT"])
+def update_ingredient():
+    json_data = request.json
+    ingredient: Ingredient = Ingredient(
+        0, json_data["name"], json_data["unit"], json_data["available"])
+    ingredient = service.addIngredient(ingredient)
+    return jsonify({
+        "status": "Success",
+        "data": {
+            "ingredient": ingredient.serialize()
+        }
+    }), 200
+
+
+@bp.route("/ingredient", methods=["DELETE"])
+def delete_ingredient():
+    json_data = request.json
+    id = json_data["id"]
+    service.deleteIngredient(id)
+    return jsonify({
+        "status": "Success",
+        "data": {
+            "id": id
+        }
+    })
```

Comment:
shouldn't we use **id** as a path parameter instead of a request param for a more restful approach? again i think the /ingredient is unnecessary cause (this controller already responds to path **/ingredients**), so maybe we should change the delete and put routes to:
```
@bp.route("/<id>", methods=["DELETE"]) 
@bp.route("/<id>", methods=["PUT"])

```
and send the id as a function param:
```
def delete_ingredient(id)
def update_ingredient(id)
```